File: CS/CustomCachedDocumentSourceSerialization/CustomPageDataService.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CustomCachedDocumentSourceSerialization {
    public class CustomPageDataService {
        public const string Key = "customPageData";
        public Dictionary<int, List<int>> PageAdditionalData { get; private set; }
        public Dictionary<int, string> SheetNames { get; private set; }
        public CustomPageDataService(Dictionary<int, List<int>> pageAdditionalData) {
            if (pageAdditionalData == null)
                throw new ArgumentNullException("pageAdditionalData");
            PageAdditionalData = pageAdditionalData;
        }

        public void PrintingSystem_XlSheetCreated(object sender, DevExpress.XtraPrinting.XlSheetCreatedEventArgs e) {
            if (PageAdditionalData.ContainsKey(e.Index)) {
                int min = PageAdditionalData[e.Index].Min();
                int max = PageAdditionalData[e.Index].Max();
                e.SheetName = string.Format("Categories_{0}-{1}", min, max);
            }
        }
    }
}

File: VB/CustomCachedDocumentSourceSerialization/CustomPageDataService.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace CustomCachedDocumentSourceSerialization
	Public Class CustomPageDataService
		Public Const Key As String = "customPageData"
		Private privatePageAdditionalData As Dictionary(Of Integer, List(Of Integer))
		Public Property PageAdditionalData() As Dictionary(Of Integer, List(Of Integer))
			Get
				Return privatePageAdditionalData
			End Get
			Private Set(ByVal value As Dictionary(Of Integer, List(Of Integer)))
				privatePageAdditionalData = value
			End Set
		End Property
		Private privateSheetNames As Dictionary(Of Integer, String)
		Public Property SheetNames() As Dictionary(Of Integer, String)
			Get
				Return privateSheetNames
			End Get
			Private Set(ByVal value As Dictionary(Of Integer, String))
				privateSheetNames = value
			End Set
		End Property
		Public Sub New(ByVal pageAdditionalData As Dictionary(Of Integer, List(Of Integer)))
			If pageAdditionalData Is Nothing Then
				Throw New ArgumentNullException("pageAdditionalData")
			End If
			Me.PageAdditionalData = pageAdditionalData
		End Sub

		Public Sub PrintingSystem_XlSheetCreated(ByVal sender As Object, ByVal e As DevExpress.XtraPrinting.XlSheetCreatedEventArgs)
			If PageAdditionalData.ContainsKey(e.Index) Then
				Dim min As Integer = PageAdditionalData(e.Index).Min()
				Dim max As Integer = PageAdditionalData(e.Index).Max()
				e.SheetName = String.Format("Categories_{0}-{1}", min, max)
			End If
		End Sub
	End Class
End Namespace